text/002-css-rules.md

```diff
@@ -0,0 +1,987 @@
+const button = rules({
+  color: "black",
+  backgroundColor: "white",
+  borderRadius: 6,
+
+  variants: {
+    color: {
+      brand: {
+        color: "#FFFFA0",
+        backgroundColor: "blueviolet",
+      },
+      accent: {
+        color: "#FFE4B5",
+        backgroundColor: "slateblue",
+      },
+    },
+    size: {
+      small: { padding: 12 },
+      medium: { padding: 16 },
+      large: { padding: 24 },
+    },
+    rounded: {
+      true: { borderRadius: 999 },
+    },
+  },
+});
+```
+
+You can use it as if you were using `css`.
+
+**Usage:**
+
+```typescript
+button({
+  color: "accent",
+  size: "large",
+  rounded: true,
+});
+```
+
+**Compiled:**
+
+```css
+.[FILE_NAME]_button__[HASH] {
+  color: black;
+  background-color: white;
+  border-radius: 6px;
+}
+
+.[FILE_NAME]_button_color_brand__[HASH] {
+  color: #ffffa0;
+  background-color: blueviolet;
+}
+.[FILE_NAME]_button_color_accent__[HASH] {
+  color: #ffe4b5;
+  background-color: slateblue;
+}
+
+.[FILE_NAME]_button_size_small__[HASH] {
+  padding: 12px;
+}
+.[FILE_NAME]_button_size_medium__[HASH] {
+  padding: 16px;
+}
+.[FILE_NAME]_button_size_large__[HASH] {
+  padding: 24px;
+}
+```
+
+## 5. Toggle Variants
+
+[Stitches's `boolean variants`](https://stitches.dev/docs/variants#boolean-variants) are a special case, but the syntax for defining them is awkward.
+
+Therefore, we introduce a specialized syntax.
+
+**Code Before:**
+
+```typescript
+const button = rules({
+  // base styles
+
+  variants: {
+    // common variants
+    rounded: {
+      true: { borderRadius: 999 },
+    },
+  },
+});
+```
+
+**Code After:**
+
+```typescript
+const button = rules({
+  // base styles
+
+  toggles: {
+    rounded: { borderRadius: 999 }
+  }
+
+  variants: {
+    // common variants
+  }
+});
+```
+
+It seems also awkward to use `true` even when using.
+
+It can be represented through an array.  
+This is also a representation isomorphism with `css()`.
+
+**Usage Before:**
+
+```typescript
+button({
+  rounded: true,
+  color: "accent",
+  size: "large",
+});
+```
+
+**Usage After:**
+
+```typescript
+button([
+  "rounded",
+  {
+    color: "accent",
+    size: "large",
+  },
+]);
+```
+
+## 6. Compound Variants
+
+[Stitches's `Compound Variants`](https://stitches.dev/docs/variants#compound-variants) is an effective way to set up additional css by leveraging the combination of variations you have already set up.
+
+However, the method of writing the conditions seems quite inconvenient.  
+So we want to improve the UX in this area.
+
+**Code Before:**
+
+```typescript
+const button = rules({
+  // base styles
+
+  variants: {
+    color: {
+      brand: { color: "#FFFFA0" },
+      accent: { color: "#FFE4B5" },
+    },
+    size: {
+      small: { padding: 12 },
+      medium: { padding: 16 },
+      large: { padding: 24 },
+    },
+  },
+  compoundVariants: [
+    {
+      color: brand,
+      size: small
+      style: {
+        fontSize: "16px",
+      },
+    },
+  ],
```

Comment:
> [!NOTE]
> **P3[Recommendations]: If you can, please do so. We encourage changes, but it's up to the developer to decide.**

I think we could also consider leaving the old way for compatibility.
What do you think?

IMO, the new way more intuitive, but the old way isn't bad either.

Reply:
Then u mean we gonna use both of ways?

Reply:
IMO, If wd need to ensure backward compatibility for users who are migrating.
In the future, a vanilla extract compatible API may be defined separately like `@mincho-js/css/compat`, but for now it's good to allow backward compatibility to ease entry.

Reply:
Ok I see, then I pick the old way together.